rsocket-transport-netty/src/main/java/io/rsocket/transport/netty/server/WebsocketServerTransport.java

```diff
@@ -114,6 +118,8 @@ public void setTransportHeaders(Supplier<Map<String, String>> transportHeaders)
             (request, response) -> {
               transportHeaders.get().forEach(response::addHeader);
               return response.sendWebsocket(
+                  null,
+                  Math.max(DEFAULT_FRAME_SIZE, mtu == 0 ? FRAME_MAX_SIZE : mtu),
```

Comment:
what is this doing? Is there a reason we are setting the max frame size to anything but rsocket max frame size?

Reply:
So, the entire fragmentation logic is built on top of the assumption that mtu is higher than 0. So if `mtu` is 0, then fragmentation is avoided. It means that max size of the fragment should be less || equal to maximum possible size frame. At least I was thinking like that when writing that code. In case 0xFFFFFF is a specific limit to fragmentation and in general payload size is unlimited (or limited to `(Integer.MAX_VALUE - 2)`), then I can fix that.

Reply:
simplified to max frame size for all possible cases